waiter/src/waiter/statsd.clj

```diff
@@ -36,21 +36,61 @@
   [host-address]
   (InetAddress/getByName host-address))
 
-(let [config (atom nil)
-      socket-agent (agent nil)
-      environment (atom nil)
-      cluster (atom nil)
-      server (atom nil)
-      publish-interval-ms (atom 0)
-      histogram-max-size (atom 10000)]
+(defn keep-all-metrics
+  "default predicate for always sending all metrics"
+  [metric-group metric]
+  true)
+
+(defn- sanitize-host [host]
+  (resolve-ip host))
+
+(defn- sanitize-port [port]
+  (if (integer? port) port (Integer/parseInt port)))
+
+(defn- sanitize-predicate [predicate-fn-symbol]
+  (if predicate-fn-symbol
+    (utils/resolve-symbol! predicate-fn-symbol)
+    keep-all-metrics))
+
+(defn make-dd-client
+  "build non-blocking dogstatsd client"
+  [{:keys [host port]} {:keys [prefix]}]
+  (NonBlockingStatsDClient.
+      ^String (or prefix "waiter")
+      ^String host
+      ^Integer (sanitize-port port)))
+
+(defn make-dd-tag-config
+  "build string arrays for common datadog tags"
+  [{:keys [cluster environment server]}]
+  (let [cluster-tags (list (str "env:" environment)
+                           (str "cluster:" cluster))
```

Comment:
Some of the other instances include the following tags:

- env: the prod/qa environment reporting the metrics/logs
- service: the component reporting the metrics/logs
- waiter-hostname: the hostname the router is running on
- waiter-cluster: the waiter cluster the router is operating as part of

Should we make these consistent?

Reply:
I don't want to go add `service` in right now (we'd have to do more refactoring to plumb that information through), but I'm happy to rename things to be consistent. `env` is already `env`, but we could change the other two:

1. `server` → `waiter-hostname`
2. `cluster` → `waiter-cluster`

Do we feel good about that? I think you (or whoever chose those names) was on the right track with making the tags more specific rather than generic. E.g., we want to be clear that "cluster" means _waiter cluster_ rather than _k8s cluster_.

Reply:
Fixed